src/utils/ExtrusionLine.cpp

```diff
@@ -27,4 +28,143 @@ void ExtrusionLine::appendJunctionsTo(LineJunctions& result) const
+     * last junctions are anyway the same.
+     * */
+    ExtrusionJunction previous_previous = junctions.front();
+    ExtrusionJunction previous = junctions.front();
+    ExtrusionJunction current = junctions.at(1);
+    ExtrusionJunction next = junctions.at(2);
+
+    /* When removing a vertex, we check the height of the triangle of the area
+     being removed from the original polygon by the simplification. However,
+     when consecutively removing multiple vertices the height of the previously
+     removed vertices w.r.t. the shortcut path changes.
+     In order to not recompute the new height value of previously removed
+     vertices we compute the height of a representative triangle, which covers
+     the same amount of area as the area being cut off. We use the Shoelace
+     formula to accumulate the area under the removed segments. This works by
+     computing the area in a 'fan' where each of the blades of the fan go from
+     the origin to one of the segments. While removing vertices the area in
+     this fan accumulates. By subtracting the area of the blade connected to
+     the short-cutting segment we obtain the total area of the cutoff region.
+     From this area we compute the height of the representative triangle using
+     the standard formula for a triangle area: A = .5*b*h
+     */
+    coord_t accumulated_area_removed = previous.p.X * current.p.Y - previous.p.Y * current.p.X; // Twice the Shoelace formula for area of polygon per line segment.
+
+    for (size_t point_idx = 1; point_idx < junctions.size() - 1; point_idx++)
+    {
+        current = junctions.at(point_idx % junctions.size());
+
+        // don't spill over if the [next] vertex will then be equal to [previous]
+        if (point_idx + 1 == junctions.size() && new_junctions.size() > 1)
+        {
+            next = new_junctions[0]; //Spill over to new polygon for checking removed area.
+        }
+        else
+        {
+            next = junctions.at(point_idx + 1);
+        }
+        const coord_t removed_area_next = current.p.X * next.p.Y - current.p.Y * next.p.X; // Twice the Shoelace formula for area of polygon per line segment.
+        const coord_t negative_area_closing = next.p.X * previous.p.Y - next.p.Y * previous.p.X; // area between the origin and the short-cutting segment
+        accumulated_area_removed += removed_area_next;
+
+        const coord_t length2 = vSize2(current - previous);
+        if (length2 < 25)
+        {
+            // We're allowed to always delete segments of less than 5 micron. The width in this case doesn't matter that
+            // much.
+            continue;
+        }
+
+        const coord_t area_removed_so_far = accumulated_area_removed + negative_area_closing; // close the shortcut area polygon
+        const coord_t base_length_2 = vSize2(next - previous);
+
+        if (base_length_2 == 0) //Two line segments form a line back and forth with no area.
```

Comment:
I am not sure if this case can occur in libArachne toolpaths at all, since libArachne should generally produce separate ExtrusionLines for cases like this. Nevertheless, I kept it in just to be sure.

Reply:
Keep in mind that eventually, your method might be called by other things than just libArachne, or Arachne might change. So keeping these things is more than sensible.